src/lib/dialog/dialog-container.ts

```diff
@@ -79,6 +79,9 @@ export class MatDialogContainer extends BasePortalOutlet {
   /** Element that was focused before the dialog was opened. Save this to restore upon close. */
   private _elementFocusedBeforeDialogWasOpened: HTMLElement | null = null;
 
+  /** Type of interaction that closed the dialog. */
+  private _closeFocusOrigin: FocusOrigin = 'program';
```

Comment:
I'd expand this description to something like
```
Type of interaction that led to the dialog being closed. This is used to determine
where the focus style will be applied when returning focus to its original location
after the dialog is closed.
```

The name `_closeInteractionType` might also be more accurate